client/test/utils.js

```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  promise: (data) => Promise.resolve().then(() => (data)),
```

Comment:
tl;dr, it doesn't work.

There's a very minor difference between Promise.resolve and normaly promises in that Promise.resolve fires on the same tick, rather than the next, so you end up with some problems if the real promise doesn't do this. So the code actually breaks with this.